--- a/src/common.h
+++ b/src/common.h
@@ -20,7 +20,7 @@
    can be found in /usr/share/common-licenses/GPL-3 file.
 */
 
-#include <asm/types.h>
+#include <linux/types.h>
 
 #ifndef _COMMON_H
 #define _COMMON_H
--- a/src/file.c
+++ b/src/file.c
@@ -34,7 +34,7 @@
 #define _LINUX_STRING_H_	/* hack to avoid inclusion of <linux/string.h> */
 #define _LINUX_FS_H		/* hack to avoid inclusion of <linux/fs.h> */
 
-#include <asm/types.h>
+#include <linux/types.h>
 
 #include <linux/msdos_fs.h>
 
--- a/src/io.c
+++ b/src/io.c
@@ -40,6 +40,7 @@
 #include <sys/ioctl.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <linux/types.h>
 #include <linux/fd.h>
 
 #include "fsck.fat.h"
--- a/src/fsck.fat.h
+++ b/src/fsck.fat.h
@@ -32,7 +32,7 @@
 #define _LINUX_STRING_H_	/* hack to avoid inclusion of <linux/string.h> */
 #define _LINUX_FS_H		/* hack to avoid inclusion of <linux/fs.h> */
 
-#include <asm/types.h>
+#include <linux/types.h>
 #include <asm/byteorder.h>
 
 #include <linux/msdos_fs.h>
--- a/src/mkfs.fat.c
+++ b/src/mkfs.fat.c
@@ -67,7 +67,7 @@
 #include <stdint.h>
 #include <endian.h>
 
-#include <asm/types.h>
+#include <linux/types.h>
 
 /* In earlier versions, an own llseek() was used, but glibc lseek() is
  * sufficient (or even better :) for 64 bit offsets in the meantime */
